src/Core/Localization/CLDR/LocaleData.php

```diff
@@ -146,7 +146,11 @@ public function overrideWith(LocaleData $localeData)
         }
 
         if (null !== $localeData->getDecimalPatterns()) {
-            $this->setDecimalPatterns($localeData->getDecimalPatterns());
+            $initialDecimalPatterns = $this->getDecimalPatterns();
```

Comment:
This is a bit weird, you added an accepted `null` type to this field So I thought why not forcing default value as empty array instead, this way the type is always respected

Then I realized that `null` is used to check if the LocaleData contains data that have to be overridden, so `null` is required but then it seems like all other fields can also be `null` so their types should also be updated right?

Finally, I see that decimal patterns is not purely overridden but rather the two values are merged which allows to have multiple values for different number systems It seems like a good idea but then two things worry me:
- the values are merged without checking for unicity so what happens if both LocaleData use `latn` number system? since it is used as the key I guess it correctly overrides it but then you should adapt you tests because you always use the same value `'#,##0.###'` so it doesn't allow to see which one is kept in the end
- if such merge behaviour is required for decimal patterns should it be generalized for other fields as well? or at least I guess should for other `array` types, I see that `currencyPatterns` does roughly the same check as you did, but `percentPatterns` doesn't, so I expect the same type of bug some day

Reply:
Other fields that perform a similar check as you did it this way:
```php
            if (null === $this->numberingSystems) {
                $this->numberingSystems = [];
            }
            $this->numberingSystems = array_merge($this->numberingSystems, $localeData->getNumberingSystems());
```

I don't care either way but they should do it the same way at least